### ISSUE TITLE
clear logger keys between lambda invocations

### ISSUE THREAD

**Issue by LironEr** [awslabs/aws-lambda-powertools-roadmap#20]
**Runtime:**  Python

**Is your feature request related to a problem? Please describe**
Currently, logger keys are kept between Lambda invocations unless I manually call `remove_keys` or `structure_logs`.

```
def handler(event, context):
  logger.info('log info') # will include key1 & key2 after first invocation

  logger.append_keys(key1=body['something'])

  logger.info('log log log')

  # some code...

  logger.append_keys(key2=some_value)

  logger.info('log log log')
```

Currently, if I want to remove all keys added during the lambda I need to run at the start of the lambda:
```
logger.remove_keys(['key1', 'key2'])
# or
logger.structure_logs()
```

The problem with the first option is that I don't want to specify all keys manually, I might even forget some keys.

The problem with the second option is that it will be removed in the next major version and it would also delete lambda injected context (if I use `@logger.inject_lambda_context`)

**Describe the solution you'd like**
An easy way to remove all keys at the start of a lambda execution
- If using `@logger.inject_lambda_context` remove all keys **by default** (can be changed with parameter) at the start of each lambda
- create a new method: `logger.restart_keys(exclude=[])`
  ```
  logger.restart_keys() # will remove all keys
  logger.restart_keys(['key2']) # will remove all keys except key2
  ```

**If you provide guidance, is this something you'd like to contribute?**
Yes

**Additional context**
Related issue: https://github.com/awslabs/aws-lambda-powertools-python/issues/407#issuecomment-853820525

**Comment by heitorlessa** [comment]
Thanks a lot for opening that up @LironEr - just added it to the backlog :)

Following our tenets, I'll implement a toggle you can use in the `@inject_lambda_context` to clear the state but not as a default, otherwise it'll break customers' current reliance on that today -- We can however make this a default in v2.

**Comment by heitorlessa** [comment]
hey @LironEr, could you double check the UX before we release it today?

https://github.com/awslabs/aws-lambda-powertools-python/pull/467

**Comment by LironEr** [comment]
Awesome, thank you!

**Comment by heitorlessa** [comment]
hey @LironEr this is now available in PyPi as part of 1.17.0: https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.17.0

Added a new section in the docs including caveats when using Layers (one of the reasons it's hard to make it the default behaviour): https://awslabs.github.io/aws-lambda-powertools-python/latest/core/logger/#clearing-all-state

Thanks a lot for helping us improve everyone's experience!
